dss-apps/dss-datamodel-center/dss-datamodel-measure-management/src/main/java/com/webank/wedatasphere/dss/datamodel/measure/service/impl/MeasureServiceImpl.java

```diff
@@ -0,0 +1,214 @@
+        }
+
+        //当更新名称时
+        if (!StringUtils.equals(vo.getName(), org.getName())) {
+            int repeat = getBaseMapper().selectCount(Wrappers.<DssDatamodelMeasure>lambdaQuery().eq(DssDatamodelMeasure::getName, vo.getName()));
+            if (repeat > 0 ) {
+                LOGGER.error("errorCode : {}, measure name can not repeat ", ErrorCode.MEASURE_UPDATE_ERROR.getCode());
+                throw new DSSDatamodelCenterException(ErrorCode.MEASURE_UPDATE_ERROR.getCode(), "measure name can not repeat ");
+            }
+        }
+
+        String orgFieldIdentifier = org.getFieldIdentifier();
+        //当更新标识时
+        if (!StringUtils.equals(vo.getFieldIdentifier(), orgFieldIdentifier)) {
+            int repeat = getBaseMapper().selectCount(Wrappers.<DssDatamodelMeasure>lambdaQuery().eq(DssDatamodelMeasure::getFieldIdentifier, vo.getFieldIdentifier()));
+
+            if (repeat > 0 || datamodelReferencService.measureReferenceCount(orgFieldIdentifier)>0) {
+                LOGGER.error("errorCode : {}, measure field identifier can not repeat or referenced", ErrorCode.MEASURE_UPDATE_ERROR.getCode());
+                throw new DSSDatamodelCenterException(ErrorCode.MEASURE_UPDATE_ERROR.getCode(), "measure field identifier can not repeat or referenced");
+            }
+        }
+        DssDatamodelMeasure updateOne = modelMapper.map(vo, DssDatamodelMeasure.class);
+        updateOne.setUpdateTime(new Date());
+        getBaseMapper().update(updateOne, Wrappers.<DssDatamodelMeasure>lambdaUpdate().eq(DssDatamodelMeasure::getId, id));
+        //同步atlas
+        UpdateModelTypeResult updateModelTypeResult = assertsSyncService.syncUpdateModel(
+                new UpdateModelEvent(this
+                        , DataModelSecurityContextHolder.getContext().getDataModelAuthentication().getUser()
+                        , vo.getFieldIdentifier()
+                        , orgFieldIdentifier
+                        , ClassificationConstant.MEASURE));
+        return 1;
+    }
+
+
+    @Override
+    @Transactional(rollbackFor = Exception.class)
+    public int deleteMeasure(Long id) throws ErrorException {
+        DssDatamodelMeasure dssDatamodelMeasure = getBaseMapper().selectById(id);
+        if (dssDatamodelMeasure == null) {
+            throw new DSSDatamodelCenterException(ErrorCode.MEASURE_DELETE_ERROR.getCode(), "measure id " + id + " not exists");
+        }
+        //校验引用情况
+        if(datamodelReferencService.measureReferenceCount(dssDatamodelMeasure.getName())>0
+                ||datamodelReferencService.measureReferenceCount(dssDatamodelMeasure.getFieldIdentifier())>0){
+            throw new DSSDatamodelCenterException(ErrorCode.MEASURE_DELETE_ERROR.getCode(), "measure id " + id + " has referenced");
+        }
+
+        getBaseMapper().deleteById(id);
+        //同步资产
+        assertsSyncService.syncDeleteModel(new DeleteModelEvent(this
```

Comment:
publisher.publishEvent(new DeleteModelEvent(this
                , DataModelSecurityContextHolder.getContext().getDataModelAuthentication().getUser()
                , dssDatamodelMeasure.getFieldIdentifier()
                , ClassificationConstant.MEASURE));